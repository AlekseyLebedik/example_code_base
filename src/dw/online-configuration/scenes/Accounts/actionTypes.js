export const ACCOUNTS_FETCH = 'accounts/ACCOUNTS_FETCH';
export const ACCOUNTS_FETCH_SUCCESS = 'accounts/ACCOUNTS_FETCH_SUCCESS';

export const ACCOUNTS_LIST_ITEM_ONCLICK = 'accounts/ACCOUNTS_LIST_ITEM_ONCLICK';
export const ACCOUNTS_EXPORT = 'accounts/ACCOUNTS_EXPORT';

export const ACCOUNTS_UPDATE_SELECTED_ACCOUNT =
  'accounts/ACCOUNTS_UPDATE_SELECTED_ACCOUNT';
