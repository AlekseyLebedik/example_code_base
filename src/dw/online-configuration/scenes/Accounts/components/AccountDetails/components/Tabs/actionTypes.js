export const ACCOUNT_DETAILS_TABS_CHANGE =
  'accounts/ACCOUNT_DETAILS_TABS_CHANGE';
