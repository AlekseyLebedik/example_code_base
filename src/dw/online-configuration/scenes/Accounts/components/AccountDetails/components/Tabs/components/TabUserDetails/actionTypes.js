export const ACCOUNTS_TAB_USER_DETAILS_FETCH =
  'accounts/ACCOUNTS_TAB_USER_DETAILS_FETCH';
export const ACCOUNTS_TAB_USER_DETAILS_FETCH_SUCCESS =
  'accounts/ACCOUNTS_TAB_USER_DETAILS_FETCH_SUCCESS';
export const ACCOUNTS_TAB_USER_DETAILS_UPDATE_PROFILE =
  'accounts/ACCOUNTS_TAB_USER_DETAILS_UPDATE_PROFILE';
export const ACCOUNTS_TAB_USER_DETAILS_UPDATE_PROFILE_SUCCESS =
  'accounts/ACCOUNTS_TAB_USER_DETAILS_UPDATE_PROFILE_SUCCESS';
export const ACCOUNTS_TAB_USER_DETAILS_DELETE_PROFILE =
  'accounts/ACCOUNTS_TAB_USER_DETAILS_DELETE_PROFILE';
export const ACCOUNTS_TAB_USER_DETAILS_CHANGE_REPUTATION =
  'accounts/ACCOUNTS_TAB_USER_DETAILS_CHANGE_REPUTATION';
export const ACCOUNTS_TAB_USER_DETAILS_CHANGE_REPUTATION_SUCCESS =
  'accounts/ACCOUNTS_TAB_USER_DETAILS_CHANGE_REPUTATION_SUCCESS';
