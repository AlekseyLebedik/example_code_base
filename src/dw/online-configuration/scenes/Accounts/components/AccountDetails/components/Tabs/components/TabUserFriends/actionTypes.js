export const ACCOUNTS_TAB_USER_FRIENDS_FETCH =
  'accounts/ACCOUNTS_TAB_USER_FRIENDS_FETCH';
export const ACCOUNTS_TAB_USER_FRIENDS_FETCH_SUCCESS =
  'accounts/ACCOUNTS_TAB_USER_FRIENDS_FETCH_SUCCESS';
