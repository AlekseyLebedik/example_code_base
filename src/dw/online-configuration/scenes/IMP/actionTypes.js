export const IMPS_FETCH = 'imp/IMPS_FETCH';
export const IMPS_FETCH_SUCCESS = 'imp/IMPS_FETCH_SUCCESS';

export const IMP_UPLOAD = 'imp/IMP_UPLOAD';
export const IMP_UPLOAD_SUCCESS = 'imp/IMP_UPLOAD_SUCCESS';

export const IMP_OPEN_UPLOAD_MODAL = 'imp/IMP_OPEN_UPLOAD_MODAL';
export const IMP_CLOSE_UPLOAD_MODAL = 'imp/IMP_CLOSE_UPLOAD_MODAL';
