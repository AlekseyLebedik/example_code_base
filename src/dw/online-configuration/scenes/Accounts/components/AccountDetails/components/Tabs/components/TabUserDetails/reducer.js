import * as AT from './actionTypes';

export const INITIAL_STATE = {
  userDetails: undefined,
  fetchedUserID: undefined,
};

export const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.ACCOUNTS_TAB_USER_DETAILS_FETCH_SUCCESS:
      return {
        ...state,
        userDetails: action.userDetails,
        fetchedUserID: action.userDetails.id,
      };
    case AT.ACCOUNTS_TAB_USER_DETAILS_CHANGE_REPUTATION_SUCCESS:
      return {
        ...state,
        userDetails: { ...state.userDetails, reputation: action.score },
      };
    default:
      return state;
  }
};
