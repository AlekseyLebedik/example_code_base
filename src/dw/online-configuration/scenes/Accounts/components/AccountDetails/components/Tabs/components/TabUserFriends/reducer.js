import * as AT from './actionTypes';

export const INITIAL_STATE = {
  userFriends: [],
  nextPageToken: undefined,
  fetchedUserID: undefined,
};

export const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.ACCOUNTS_TAB_USER_FRIENDS_FETCH:
      return {
        ...state,
        fetchedUserID: action.userID,
      };
    case AT.ACCOUNTS_TAB_USER_FRIENDS_FETCH_SUCCESS:
      return {
        ...state,
        userFriends: action.append
          ? [...state.userFriends, ...action.userFriends]
          : action.userFriends,
        nextPageToken: action.nextPageToken,
      };
    default:
      return state;
  }
};
