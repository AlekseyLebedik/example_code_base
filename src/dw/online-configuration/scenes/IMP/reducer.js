import * as AT from './actionTypes';

const INITIAL_STATE = {
  data: [],
  nextPageToken: undefined,
  impUploadModalVisible: false,
  impUploadModalLoading: false,
};

export const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.IMPS_FETCH_SUCCESS:
      return {
        ...state,
        data: action.append ? [...state.data, ...action.data] : action.data,
        nextPageToken: action.nextPageToken,
        q: action.q,
      };
    case AT.IMP_OPEN_UPLOAD_MODAL:
      return {
        ...state,
        impUploadModalVisible: true,
      };
    case AT.IMP_UPLOAD:
      return {
        ...state,
        impUploadModalLoading: true,
      };
    case AT.IMP_CLOSE_UPLOAD_MODAL:
    case AT.IMP_UPLOAD_SUCCESS:
      return {
        ...state,
        impUploadModalVisible: false,
        impUploadModalLoading: false,
      };
    default:
      return state;
  }
};
