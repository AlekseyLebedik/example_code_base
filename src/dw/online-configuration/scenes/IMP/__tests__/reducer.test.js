import { reducer } from '../reducer';
import * as AT from '../actionTypes';

describe('IMP', () => {
  describe('Reducer', () => {
    const initialState = {
      ...reducer(undefined, {}),
      data: [
        { id: 1, status: 'applied' },
        { id: 2, status: 'failed' },
      ],
    };

    it('returns default state', () => {
      const state = reducer(undefined, {});
      expect(state).toMatchSnapshot();
    });

    describe('IMPS_FETCH_SUCCESS', () => {
      const newEntries = [
        { id: 300, status: 'failed' },
        { id: 400, status: 'applied' },
      ];

      it('replaces the entries with new ones when `append` is false, and update list metadata', () => {
        const action = {
          type: AT.IMPS_FETCH_SUCCESS,
          append: false,
          data: newEntries,
          nextPageToken: 'YYZ',
        };
        const state = reducer(initialState, action);
        expect(state).toMatchSnapshot();
      });

      it('combines both entry lists when `append` is true, and update list metadata', () => {
        const action = {
          type: AT.IMPS_FETCH_SUCCESS,
          append: true,
          data: newEntries,
          nextPageToken: 'YYZ',
        };
        const state = reducer(initialState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('IMP_OPEN_UPLOAD_MODAL', () => {
      it('sets modal visibility to true', () => {
        const action = {
          type: AT.IMP_OPEN_UPLOAD_MODAL,
        };
        const state = reducer(initialState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('IMP_UPLOAD', () => {
      it('sets modal loading to true', () => {
        const action = {
          type: AT.IMP_UPLOAD,
        };
        const state = reducer(initialState, action);
        expect(state).toMatchSnapshot();
      });
    });

    const modalState = {
      ...initialState,
      impUploadModalVisible: true,
      impUploadModalLoading: true,
    };

    describe('IMP_CLOSE_UPLOAD_MODAL', () => {
      it('sets modal visibility and loading to false', () => {
        const action = {
          type: AT.IMP_CLOSE_UPLOAD_MODAL,
        };
        const state = reducer(modalState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('IMP_UPLOAD_SUCCESS', () => {
      it('sets modal visibility and loading to false', () => {
        const action = {
          type: AT.IMP_UPLOAD_SUCCESS,
        };
        const state = reducer(modalState, action);
        expect(state).toMatchSnapshot();
      });
    });
  });
});
