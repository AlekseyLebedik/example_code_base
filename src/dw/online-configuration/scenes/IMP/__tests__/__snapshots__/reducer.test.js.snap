// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IMP Reducer IMP_CLOSE_UPLOAD_MODAL sets modal visibility and loading to false 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "status": "applied",
    },
    Object {
      "id": 2,
      "status": "failed",
    },
  ],
  "impUploadModalLoading": false,
  "impUploadModalVisible": false,
  "nextPageToken": undefined,
}
`;

exports[`IMP Reducer IMP_OPEN_UPLOAD_MODAL sets modal visibility to true 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "status": "applied",
    },
    Object {
      "id": 2,
      "status": "failed",
    },
  ],
  "impUploadModalLoading": false,
  "impUploadModalVisible": true,
  "nextPageToken": undefined,
}
`;

exports[`IMP Reducer IMP_UPLOAD sets modal loading to true 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "status": "applied",
    },
    Object {
      "id": 2,
      "status": "failed",
    },
  ],
  "impUploadModalLoading": true,
  "impUploadModalVisible": false,
  "nextPageToken": undefined,
}
`;

exports[`IMP Reducer IMP_UPLOAD_SUCCESS sets modal visibility and loading to false 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "status": "applied",
    },
    Object {
      "id": 2,
      "status": "failed",
    },
  ],
  "impUploadModalLoading": false,
  "impUploadModalVisible": false,
  "nextPageToken": undefined,
}
`;

exports[`IMP Reducer IMPS_FETCH_SUCCESS combines both entry lists when \`append\` is true, and update list metadata 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "status": "applied",
    },
    Object {
      "id": 2,
      "status": "failed",
    },
    Object {
      "id": 300,
      "status": "failed",
    },
    Object {
      "id": 400,
      "status": "applied",
    },
  ],
  "impUploadModalLoading": false,
  "impUploadModalVisible": false,
  "nextPageToken": "YYZ",
  "q": undefined,
}
`;

exports[`IMP Reducer IMPS_FETCH_SUCCESS replaces the entries with new ones when \`append\` is false, and update list metadata 1`] = `
Object {
  "data": Array [
    Object {
      "id": 300,
      "status": "failed",
    },
    Object {
      "id": 400,
      "status": "applied",
    },
  ],
  "impUploadModalLoading": false,
  "impUploadModalVisible": false,
  "nextPageToken": "YYZ",
  "q": undefined,
}
`;

exports[`IMP Reducer returns default state 1`] = `
Object {
  "data": Array [],
  "impUploadModalLoading": false,
  "impUploadModalVisible": false,
  "nextPageToken": undefined,
}
`;
